forms/signals.py

```diff
@@ -0,0 +1,29 @@
+import os
+
+from django.db import models
+from django.dispatch import receiver
+
+from forms.models.form import Attachment
+
+
+@receiver(models.signals.post_delete, sender=Attachment)
+def auto_delete_file_on_delete(sender, instance, **kwargs):
+    if instance.file:
+        if os.path.isfile(instance.attachment.path):
+            os.remove(instance.attachment.path)
+
+
+@receiver(models.signals.pre_save, sender=Attachment)
+def auto_delete_file_on_change(sender, instance, **kwargs):
+    if not instance.pk:
+        return False
+
+    try:
+        old_file = Attachment.objects.get(pk=instance.pk).attachment
+    except Attachment.DoesNotExist:
+        return False
+
+    new_file = instance.attachmentM
```

Comment:
Would be good to write test which confirms the delete after PATCH request? This typo would have spotted there. I think generally it's important to test Signals, because those are doing all kinds of side effects outside of the "normal route".

Reply:
And ofc test that the file has gone to valhalla after DELETE request also...